### PR TITLE
prov/tcp move iovec adjust function to common code 

### PR DESCRIPTION
**Pull request by vkrishna**

### CODE REVIEW

File: prov/tcp/src/tcpx_progress.c [hunk 1]
```diff
@@ -134,7 +134,7 @@ static void process_tx_pe_entry(struct tcpx_pe_entry *pe_entry)
 	int ret;
 
 	ret = tcpx_send_msg(pe_entry);
-	if (ret && OFI_SOCK_TRY_SND_RCV_AGAIN(ret))
+	if (ret && OFI_SOCK_TRY_SND_RCV_AGAIN(-ret))
```
Comment (shefty) [inline prov/tcp/src/tcpx_progress.c]: This can just be
```
if (OFI_SOCK_TRY_SND_RCV_AGAIN(-ret))
```

File: src/iov.c [hunk 2]
```diff
@@ -67,3 +67,25 @@ uint64_t ofi_copy_iov_buf(const struct iovec *iov, size_t iov_count, uint64_t io
...
+
+		new_iov = i+1;
+		buf_offset = offset - len;
+	}
+
+	iovec[new_iov].iov_base = ((uint8_t *)(iovec[new_iov].iov_base) +
+				   buf_offset);
+	iovec[new_iov].iov_len -= buf_offset;
+	(*iovec_count) -= new_iov;
+}
```
Comment (shefty) [inline src/iov.c]: This function doesn't look correct., or I'm not understanding what it's attempting to do.  The following code is closer to what I think is needed (with an optimization for the single iovec case - renames to match similar code in this file):
```
void ofi_consume_iov(struct iovec *iov, size_t *iov_count, size_t consumed)
{
    size_t i;

    if (*iov_count == 1)
        goto out;

    for (i = 0; i < *iov_count; i++) {
        if (consumed < iov[i].iov_len)
            break;
        consumed -= iov[i].iov_len;
    }
    memmove(iov, &iov[i], sizeof(*iov) * (*iov_count - i));
    *iov_count -= i;
out:
    iov[0].iov_base += consumed;
    iov[0].iov_len -= consumed;
}
```
I don't see a memmove type operation in the above code that would 'shift' the iovec entries forward.  Either that's missing or I don't understand what the intent of this function is.

Comment (vkrishna) [reply src/iov.c]: Yes the memmove should be there. I will just use your function. thanks

**Comment by shefty** [comment]
@a-ilango @aingerson - Please take a look at the new function ofi_consume_iov() being introduced here.  It may be useful for implementing multi-recv.

**Comment by aingerson** [comment]
I thought you said multi recv was only allowed for one iov?

**Comment by shefty** [comment]
@a-ilango @aingerson - Please take a look at the new function ofi_consume_iov() being introduced here.  It may be useful for implementing multi-recv.

**Comment by shefty** [comment]
It is only allowed for one.  So this call may not be that helpful, except to remove maybe 2 lines of code.  :)

**Comment by a-ilango** [comment]
> Please take a look at the new function ofi_consume_iov() being introduced here. It may be useful for implementing multi-recv

Sure. How the new iov is being generated is slightly different but I'll take a look.

**Pull request merged**